MetaMask/metamask-mobile
# chore: remove remaining use of Preferences Controller for account information

## **Description**

1. What is the reason for the change?
- In previous issues we migrated all account information state to reference the the new accounts controller redux state. - This change furthers this effort and removes all remaining references to PreferencesController.identities and PreferencesController.selectedAddress. 
- All of the remaining references are non state values and thus are not going to be using the redux state implemented in the previous issues. Instead we will reference the Engine.context.AccountsController instance directly.
- This PR also removes `selectSelectedAddress` and `selectIdentities` from the available redux selectors to prevent future use of this data.
    - This is ideal since we eventually want to completely remove this data from the PreferencesController in favour of the AccountsController
- After this PR there should be `zero references to this data in the codebase`.

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/537

## **Review checklist**
- [x] confirm that there are no reference to `PreferencesController.identities` in the codebase
- [x] Confirm that there are no references to `PreferencesController.selectedAddress` in the codebase
- [x] Confirm that there are no references to `selectIdentities`
- [x] Confirm that there are no references to `selectSelectedAddress`

(There will still be use of these variables in the migration files that are migrating this data from PreferencesController to AccountsController. There should be no other references besides the migrations)

## **Manual testing steps**

1. Import a wallet ideally with funds
2. after onboarding, create a new account
5. edit then name of this new account
6. add a hardware wallet via ledger or QR
7. import a private key like orangefox.eth or another of your choosing
8. go through the send flow from an account with funds
9. EXPECT, all addresses, names and transaction data should match the account listed in the transaction
10. EXPECT, all addresses to be in checksum format (example 0xC4955C0d639D99699Bfd7Ec54d9FaFEe40e4D272 vs 0xc4955c0d639d99699bfd7ec54d9fafee40e4d272)
11. navigate to https://metamask.github.io/test-dapp/ in the mobile browser
12. connect one or more account
13. EXPECT, the connection modal should display the correct account information as well as the address in checksum format
14. perform a send on the test dapp
15. EXPECT, the send modal should display the correct account information and the address in checksum format
16. Disconnect the account from the test dapp
17. reconnect the account vai wallet connect
18. EXPECT, the connection should work and should display the correct account information as well as the address in checksum format

## **Screenshots/Recordings**


### **Before**


### **After**

<img width="639" alt="Screenshot 2024-07-24 at 3 15 51 PM" src="https://github.com/user-attachments/assets/97a7c262-f95a-4343-bfcf-d0ae4e1c6aff">
<img width="679" alt="Screenshot 2024-07-24 at 3 16 02 PM" src="https://github.com/user-attachments/assets/df88ce65-139c-4e54-9a60-a86ba9df5ee9">
<img width="801" alt="Screenshot 2024-07-24 at 3 16 37 PM" src="https://github.com/user-attachments/assets/0cd26e47-8073-4000-bb85-73a4de372450">
<img width="666" alt="Screenshot 2024-07-24 at 3 20 00 PM" src="https://github.com/user-attachments/assets/7a8fd678-414a-45ab-a8e9-05083f5fb138">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
